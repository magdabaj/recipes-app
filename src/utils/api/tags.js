import "babel-polyfill"

const api = 'https://afternoon-plateau-23579.herokuapp.com/tags';
// const api = 'http://localhost:5000/tags'

export const fetchAllTags = async () => {
    const response = await fetch(api);
    const data = await response.json();
    if (response.status >= 400) {
        throw new Error(data.errors);
    }
    return data;
};