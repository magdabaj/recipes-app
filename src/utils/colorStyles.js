const colorStyles = {
    darkRed: '#5D001E',
    lightGrey: '#E3E2DF',
    lightPink: '#E3AFBC',
    mediumPink: '#EE4C7C',
    darkPink: '#9A1750',
};

export default colorStyles;
