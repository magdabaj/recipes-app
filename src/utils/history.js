// import { createBrowserHistory } from 'history';
//
// const history = createBrowserHistory();

import { createMemoryHistory } from "history";

const history = createMemoryHistory()

export default history;
