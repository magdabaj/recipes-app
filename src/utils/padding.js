const basePadding = 8;

const padding = {
    extraSmall: `${basePadding / 4}px`, // 2px
    small: `${basePadding / 2}px`, // 4px
    base: `${basePadding}px`, // 8px
    medium: `${basePadding * 1.5}px`, // 12px
    large: `${basePadding * 2}px`, // 16px
}

export default padding;