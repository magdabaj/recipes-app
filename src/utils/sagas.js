import { all } from 'redux-saga/effects';
import { watchRecipesByTagLoad, watchRecipesLoad, watchTagsLoad } from "../containers/HomePage/saga";

export default function* rootSaga() {
    yield all([
        watchRecipesLoad(),
        watchTagsLoad(),
        watchRecipesByTagLoad()
    ]);
}