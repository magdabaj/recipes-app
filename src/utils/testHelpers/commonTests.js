const commonTests = (render, useFirstChildAsSnapshot = true) => {
    it('does not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error')
        render()
        expect(spy).not.toHaveBeenCalled()
    })

    it('should render and match the snapshot',  () => {
        const { container } = render()
        expect(container).not.toBeNull()
        if(useFirstChildAsSnapshot) expect(container.firstChild).not.toBeNull()
        expect(
            useFirstChildAsSnapshot ? container.firstChild : container,
        ).toMatchSnapshot()
    });
}

export default commonTests;