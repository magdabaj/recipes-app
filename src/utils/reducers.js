import { combineReducers } from 'redux';
import { connectRouter } from "connected-react-router";
import homePageReducer from "../containers/HomePage/reducer";
import loginContainerReducer from "../containers/LoginContainer/reducer";
import userRecipesContainerReducer from "../containers/UserRecipesContainer/reducer";
import recipesFormContainerReducer from "../containers/RecipesFormContainer/reducer";
import recipeContainerReducer from "../containers/RecipeContainer/reducer";
import appReducer from '../containers/App/reducer';
import history from "./history";

const rootReducer = combineReducers({
    homePageReducer,
    loginContainerReducer,
    userRecipesContainerReducer,
    recipesFormContainerReducer,
    recipeContainerReducer,
    appReducer,
    router: connectRouter(history),
})

export default rootReducer;