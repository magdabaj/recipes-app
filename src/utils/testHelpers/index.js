import { Router } from 'react-router-dom'
import { createMemoryHistory } from "history";
import store from "../store";
import {Provider} from "react-redux";
import React from "react";
import { render } from "@testing-library/react";
import {createStore} from "redux";
// import {reducer} from "react-toastify/dist/hooks/toastContainerReducer";

// const store  = createStore(rootReducer, initialState={})
const memoryHistory = createMemoryHistory({
    initialEntries: ['/'],
})

export const withStoreAndRouter = component =>
    <Router history={memoryHistory}>
        <Provider store={store}>
            {component}
        </Provider>
    </Router>

export const renderWithRouter = (
    ui,
    {
        history = memoryHistory,
    } = {}
) => (
    {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
)

export const renderWithRouterKentCDodds = (
    ui,
        {
            route='/',
            history = createMemoryHistory({initialEntries: [route]}),
            ...renderOptions
        } = {}
) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Router history={history}>
            {children}
        </Router>
    }

    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions})
    }
}

export const renderWithProviderKentCDodds = (
        ui,
        reducer,
        {initialState, store = createStore(reducer, initialState), ...options} = {},
    ) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
    return render(ui, {wrapper: Wrapper, ...options})
}