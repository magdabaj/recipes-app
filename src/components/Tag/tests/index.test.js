import {renderWithRouter} from "../../../utils/testHelpers";
import { screen } from "@testing-library/dom";
import TagComponent from "../index";
import {fakeTags} from "../../../utils/testHelpers/fixtures/tags";
import React from "react";
import commonTests from "../../../utils/testHelpers/commonTests";
import userEvent from "@testing-library/user-event";

const renderTagComponent = () =>
    renderWithRouter(<TagComponent tags={fakeTags} tagType={'alkohol'}/>)

commonTests(renderTagComponent)

test('shows tagValues after on click', () => {
    const {history} = renderTagComponent()

    const tagType = screen.getByText(/alkohol/i)

    userEvent.click(tagType)
    const tagValue1 = screen.getByText(/rum/i)
    const tagValue2 = screen.getByText(/wino/i)
    expect(tagValue1).toBeInTheDocument()
    expect(tagValue2).toBeInTheDocument()

    userEvent.click(tagValue2)
    expect(history.location.pathname).toBe('/tag/2')
})