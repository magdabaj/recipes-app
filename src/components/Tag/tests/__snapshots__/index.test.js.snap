// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<li
  class="TagTypeContainer-u54ots-0 dJalNb"
>
  <span
    class="tag-type"
  >
    alkohol
  </span>
  <ul
    class="UlComponent-z53gs3-0 gLCPYA"
  />
</li>
`;
